[Feature]: battle create api

U-S-jun opened the pull request:
## 🔗 관련 이슈
<!-- 예: Closes #123, Relates to #456 -->
- #9 

## ✅ 작업 내용

### 💡개선 사항
<!-- 이 PR에서 변경된 주요 내용 개선 사항 -->

`battle.service.ts` : 배틀 객체 생성 후 배틀 배열에 추가 후 생성된 배열을 return 합니다.
`battle.controller.ts` : 배틀 생성 요청에 대해 body 검증을 하고 배틀 생성 후 id를 return 합니다.

<br />

## 📸 참고 사항
<!-- 코드 리뷰에서 참고할 사항, 새로 주의 깊게 봐야 하는 파일/코드, 주요 고민과 해결 과정-->

Redis 도입 전 이므로 인메모리 battle 배열에 battle 생성 후 push 했습니다.
API 명세서를 참고하여 요청/응답을 검증, 구성했습니다.

<br />

## 💬 질문사항 (고민했던 부분)
<!-- 리뷰어에게 묻고 싶은 것, 트레이드오프 등 -->

U-S-jun committed:
feature(be): 배틀 생성 라우팅 및 요청 body 검증

U-S-jun committed:
feature(be): 타입 생성

U-S-jun committed:
feature(be): 배틀 생성 서비스 구현

U-S-jun committed:
test(be): 배틀 생성 단위테스트 작성

U-S-jun committed:
feature(be): 모듈 추가

HoonDongKang commented on backend/src/battles/battles.controller.ts:
```diff
@@ -0,0 +1,71 @@
+    return { battleId: battle.id }
+  }
+
+  private validateCreateBattleRequest(body: CreateBattleRequest): string[] {
```
`DTO`에서 `class-validator`로 검사를 진행할 수 있는데, validation pipe는 사용하지 않으실 생각이신가요??

HoonDongKang commented on backend/src/battles/battles.service.ts:
```diff
@@ -0,0 +1,42 @@
+
+  create(payload: CreateBattleRequest): Battle {
+    const now = new Date()
+    const status: BattleStatus = 'PENDING'
```
`BATTLE_STATUS.PENDING` 이런 상수로 관리하시는 건 어떠신가요?

HoonDongKang commented on backend/src/battles/battles.service.ts:
```diff
@@ -0,0 +1,42 @@
+      category: payload.category.trim(),
+      playTime: payload.playTime,
+      isPublic,
+      password: isPublic ? undefined : payload.password?.trim(),
```
`type` 값을  `PUBLIC`, `PRIVATE`으로 받아오는 것으로 이해했어서, `isPublic` 데이터와 중복되는 것 같습니다! 
하단에 `password` 또한 `type` 값으로 비교할 수 있을 것 같아요.

HoonDongKang reviewed (approved):
고생하셨습니다!

jjjjjinseo commented on backend/src/app.module.ts:
```diff
@@ -1,9 +1,10 @@
 
 @Module({
-  imports: [],
+  imports: [BattlesModule],
```
저희 클래스 이름에도 Battles.. 이렇게 복수형으로 하면 좋을까요?

jjjjjinseo commented on backend/src/battles/battles.controller.spec.ts:
```diff
@@ -0,0 +1,49 @@
+import { BattlesController } from './battles.controller'
+import { BattlesService } from './battles.service'
+
+describe('BattlesController', () => {
```
앗 테스트까지 진행해주셨군요 💯

jjjjjinseo commented on backend/src/battles/battles.service.ts:
```diff
@@ -0,0 +1,42 @@
+    const isPublic = !payload.password
+
+    const battle: Battle = {
+      id: randomUUID(),
```
UUID 도입 좋은 것 같습니다!

HoonDongKang replied:
아 이 부분도 리뷰해보려했는데 `randomUUID`는 v4라서, timestamp로 정렬이 가능한 v7은 어떠신가요?

jjjjjinseo commented on backend/src/battles/types.ts:
```diff
@@ -0,0 +1,40 @@
+
+export type BattlePhase = 'WAITING_FOR_START' | 'OPINION_SHARE' | 'TEAM_A_ATTACK' | 'TEAM_B_ATTACK' | 'TEAM_SWITCH'
+
+export interface CreateBattleRequest {
```
DTO는 dto 폴더에 따로 하는거 어떨까요?

jjjjjinseo reviewed (commented):
수고하셨습니다 : )

U-S-jun committed:
feature(be): 리뷰 반영

상수 관리 : BATTLE_STATUS
validation pipe 도입
isPublic 삭제
UUIDv7 도입
DTO 폴더 관리
폴더 구조 컨벤션 반영

U-S-jun committed:
feature(be): validation pipe

U-S-jun committed:
의존성 추가

U-S-jun committed:
의존성 추가

rhxfkg reviewed (approved):
새벽까지 하신것 같던데 수고하셨습니당!
